main.cc

```diff
@@ -121,6 +122,82 @@ static void SetVar(StringPiece l, VarOrigin origin) {
 
 extern "C" char** environ;
 
+class SegfaultHandler {
+ public:
+  explicit SegfaultHandler(Evaluator* ev);
+  ~SegfaultHandler();
+
+  void handle(int, siginfo_t*, void*);
+
+ private:
+  static SegfaultHandler* global_handler;
+
+  void dumpstr(const char* s) const {
+    (void)write(STDERR_FILENO, s, strlen(s));
+  }
+  void dumpint(int i) const {
+    char buf[11];
+    snprintf(buf, sizeof(buf), "%d", i);
```

Comment:
I'm fairly sure sprintf with "%d" is async signal safe, but have you double-checked? Maybe worth a comment or 
even writing your own simple itoa.

Reply:
Doesn't look like it's guaranteed. I just switched this to a simple itoa implementation.